sys/include/net/gnrc/tcp.h

```diff
@@ -40,11 +40,13 @@ extern "C" {
  */
 typedef struct {
     int family;                            /**< IP address family. */
-#ifdef MODULE_GNRC_IPV6
     union {
+#ifdef MODULE_GNRC_IPV6
         uint8_t ipv6[sizeof(ipv6_addr_t)]; /**< IPv6 address storage */
-    } addr;                                /**< IP address storage */
+#else
```

Comment:
```suggestion
#endif
```

This way it is better extensible ;-) (of course the `#endif` below needs to be removed)